### PR TITLE
Fixed behavior for selected elements

### LINKED ISSUE #920: Optimize by not outputting rect mask in each case
The default rect mask is not needed for output in many/most of cases.

I can be convinced otherwise, but I'd also challenge the need to have these default masks in the data model. It's totally ok to create a mask on the fly for display/frame components as needed for dnd or other features. But our data model does not necessarily need to follow each aspect of incidental technical requirements. The closer the model to what the user can actually "input" or affect manually the better.

### PR DESCRIPTION
**Pull request by wassgha**
Closes #920

### Changes
- While dragging, only hide selection box for maskable elements (fixes behavior where text selection box was hidden and snap lines appeared out of context)
- Hide selection box for drop targets to clearly show which elements are drop targets
- Removes need for rectangular masks in output (#920)
- Minor cleanups and consolidation of variables

### CODE REVIEW

File: assets/src/edit-story/components/canvas/useInsertElement.js [hunk 1]
```diff
@@ -162,11 +162,15 @@ function createElementForCanvas(
...
-    },
+    ...(resource
+      ? {
+          resource: {
+            ...resource,
+            width,
+            height,
+          },
+        }
+      : {}),
```
Comment (dvoytenko) [inline assets/src/edit-story/components/canvas/useInsertElement.js]: nit: nulls/undefines/false/etc - work just fine for this.

File: assets/src/edit-story/masks/output.js [hunk 2]
```diff
@@ -81,13 +81,9 @@ export default function WithMask({
...
-        ...(!isBackground
-          ? {
-              clipPath: `url(#${maskId})`,
-              // stylelint-disable-next-line property-no-vendor-prefix
-              WebkitClipPath: `url(#${maskId})`,
-            }
-          : {}),
+        clipPath: `url(#${maskId})`,
+        // stylelint-disable-next-line property-no-vendor-prefix
+        webkitClipPath: `url(#${maskId})`,
```
Comment (dvoytenko) [inline assets/src/edit-story/masks/output.js]: Ok. I'm totally confused. Is it "W" or "w"? Both?

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: assets/src/edit-story/masks/output.js (hunk 2 above)
Comment (wassgha) [reply assets/src/edit-story/masks/output.js]: It's `WebkitClipPath` but the linter keeps changing it back...

Comment (wassgha) [reply assets/src/edit-story/masks/output.js]: Disabled stylelint for that line for now

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**